### PR TITLE
feat: resolve relative paths

### PR DESCRIPTION
**Pull request by kasvith**

### CODE REVIEW

**Comment by kasvith** [comment]
Closes #2

**Comment by userquin** [comment]
The resolve path to locate the images was fine, the issue was to use the base URL path,  and make it working using the New path URL relative to the base root.

There are 2 cases to solve:
- if the astro Page is under /foo/ folder, all img src should be relative to the root path url: rn Will not work since images src Will have `content/....` (issue 2)
- if we configure vite base path to /foo/ the app Will be under this context path

If vite base path is relative (vite 3 by default is `./`) we cannot use vite base as the default prefix.  So we need to make all img src paths relative to the importer,  and this can be a problem, since we need to Know if the md is being used as a Page or from an import from another Page (our use case).

**Comment by kasvith** [comment]
I tried creating a folder for a new astro page like following with same content we had for index

```
src/pages/foo/index.astro
```

This correctly displayed images in dev mode.
image

File: astro.config.ts [hunk 1]
```diff
@@ -44,11 +45,18 @@ export default defineConfig({
...
                       if (matcher) {
                         let newCode = code
-                        const path = dirname(relative(root, id)).replace(/\\/g, '/')
+                        const fileRootDir = dirname(id)
+
                         do {
-                          images.push(`${path}/${matcher[2]}`)
-                          newCode = newCode.replaceAll(matcher[1], `${path}/${matcher[2]}`)
-                          newCode = newCode.replaceAll(`(${matcher[2]})`, `(${path}/${matcher[2]})`)
+                          const assetPath = resolve(fileRootDir, matcher[2])
```
Comment (userquin) [inline astro.config.ts]: move this declaration outside the loop: 
```ts
let assetsPath: string
```

Comment (kasvith) [reply astro.config.ts]: but this depends on the file from `matcher[2]`

we are resolving `matcher[2]` based on currently processed markdown file's absolute location

Comment (userquin) [reply astro.config.ts]: we don't need a new variable on each iteration, just move the declaration outside the loop, in the loop just assign the value to it

Comment (kasvith) [reply astro.config.ts]: done

Comment (userquin) [reply astro.config.ts]: add the type, we're using ts: `let assetPath: string`

later I'll add 2 scripts for lint

Comment (kasvith) [reply astro.config.ts]: done

**Review by userquin — APPROVED** [review]

**Pull request merged**